sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java

```diff
@@ -24,9 +24,12 @@
 
 /**
  * @since 1.10
+ * @deprecated since 5.2 there's no more decorator on batch side
+ * TODO add link to no API
  */
 @BatchSide
 @ExtensionPoint
+@Deprecated
 public interface Decorator extends CheckProject {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)